Shared/Samples/Show device location with NMEA data sources/ShowDeviceLocationWithNMEADataSourcesView.swift

```diff
@@ -0,0 +1,153 @@
+    @State private var isShowingAlert = false
+    
+    /// An error from the `EAAccessoryManager`.
+    @State private var accessoryError: AccessoryError? {
+        didSet { isShowingAlert = accessoryError != nil }
+    }
+    
+    /// An error from starting the location data source.
+    @State private var locationDataSourceError: Error? {
+        didSet { isShowingAlert = locationDataSourceError != nil }
+    }
+    
+    var body: some View {
+        MapView(map: model.map)
+            .locationDisplay(model.locationDisplay)
+            .overlay(alignment: .top) {
+                VStack(alignment: .leading) {
+                    Text(model.accuracyStatus)
+                    Text(model.satelliteStatus)
+                }
+                .frame(maxWidth: .infinity)
+                .padding(.vertical, 6)
+                .background(.thinMaterial, ignoresSafeAreaEdges: .horizontal)
+            }
+            .toolbar {
+                ToolbarItemGroup(placement: .bottomBar) {
+                    Menu("Source") {
+                        Button("Mock Data") {
+                            Task {
+                                do {
+                                    try await model.start(usingMockedData: true)
+                                } catch {
+                                    self.locationDataSourceError = error
+                                }
+                            }
+                        }
+                        Button("Device") {
+                            Task {
+                                do {
+                                    try selectDevice()
+                                    try await model.start()
+                                } catch let error as AccessoryError {
+                                    self.accessoryError = error
+                                } catch {
+                                    self.locationDataSourceError = error
+                                }
+                            }
+                        }
+                    }
+                    .disabled(model.isSourceMenuDisabled)
+                    Spacer()
+                    Button("Recenter") {
+                        model.locationDisplay.autoPanMode = .recenter
+                    }
+                    .disabled(model.isRecenterButtonDisabled)
+                    Spacer()
+                    Button("Reset") {
+                        Task {
+                            await model.reset()
+                        }
+                    }
+                    .disabled(model.isResetButtonDisabled)
+                }
+            }
+            .alert("Error", isPresented: $isShowingAlert, presenting: accessoryError) { _ in
+                EmptyView()
+            } message: { error in
+                Text(error.detail)
+            }
+            .alert(isPresented: $isShowingAlert, presentingError: locationDataSourceError)
```

Comment:
Both of these alerts present when `isShowingAlert` is true. Should each alert have their own respective `isPresented` Boolean value?

Reply:
Yes and no.
From a coding best practice perspective, each error should be presented separately (even to create a custom UI to avoid an error being dismissed https://github.com/Esri/arcgis-maps-sdk-swift-toolkit/issues/315)
In this sample, the errors won't be thrown simultaneously - only 1 error will be thrown for either device or mock data.